### PR TITLE
strict, remove setindex, remove permutedims, add some docs

### PR DESCRIPTION
**Pull request by KristofferC**
* Run `makedocs` in strict mode so failing doctests fails the build.
* Remove setindex and document what to do instead.
* Remove permutedims and put it as a test helper instead.
* Add docs for dev.
* Add slice for Vec.

### CODE REVIEW

**Comment by KristofferC** [comment]
Builds fail now because the docstets fail on 32 bit windows. Nice.

File: docs/src/man/indexing.md [hunk 1]
```diff
@@ -21,13 +21,36 @@ julia> B[1, 2, 1, 2]
...
+Since `Tensor`s are immutable there is no `setindex!` function defined on them. Instead, use the functionality to create tensors from functions as described [here](@ref function_index). As an example, this sets the `[1,2]` index on a tensor to one and the rest to zero:
+
+```jldoctest
+julia> Tensor{2, 2}((i,j) -> i == 1 && j == 2 ? 1.0 : 0.0)
+2×2 ContMechTensors.Tensor{2,2,Float64,4}:
+ 0.0  1.0
+ 0.0  0.0
+```
+
+For symmetric tensors, note that you should only set the lower triangular part of the tensor:
```
Comment (fredrikekre) [inline docs/src/man/indexing.md:44]: This is weird. Should we add a check like [this](https://github.com/KristofferC/ContMechTensors.jl/blob/master/src/indexing.jl#L5) when creating with functions?

File: docs/src/man/other_operators.md [hunk 2]
```diff
@@ -101,14 +101,24 @@ majorsymmetric
...
 
 The skew symmetric part of a symmetric tensor is zero.
 
 ```@docs
 skew
 ```
 
+## Deviator
+
+The deviatorc part of a second order tensor is defined by
```
Comment (fredrikekre) [inline docs/src/man/other_operators.md]: deviatoric

File: docs/src/man/other_operators.md [hunk 3]
```diff
@@ -101,14 +101,24 @@ majorsymmetric
...
+
+$\mathbf{A}^\text{dev} = \mathbf{A} - \frac{1}{3} \mathrm{tr} \mathbf{A}$
```
Comment (fredrikekre) [inline docs/src/man/other_operators.md]: times unit tensor at the end. Maybe add this too: A_{ij}^\text{dev} = A_{ij} - 1/3 \* A_{kk}\delta_{ij}

**Comment by fredrikekre** [comment]
> Add slice for Vec.

This removes the option to make a Julia `Array` of a `Vec` by using `Colon`:

master:

``` jl
julia> A = rand(Vec{3});

julia> A[:]
3-element Array{Float64,1}:
 0.768448
 0.940515
 0.673959
```

PR:

``` jl
julia> A = rand(Vec{3});

julia> A[:]
3-element ContMechTensors.Tensor{1,3,Float64,3}:
 0.768448
 0.940515
 0.673959
```

Also, the behaviour of `Colon` for `Vec` and `Tensor` becomes different, since we can make a `Vector` of a `Tensor` like this:

``` jl
julia> B = rand(Tensor{2,3});

julia> B[:]
9-element Array{Float64,1}:
 0.768448 
 0.940515 
 0.673959 
 0.395453 
 0.313244 
 0.662555 
 0.586022 
 0.0521332
 0.26864
```

Should we error out `getindex(Tensor, ::Colon)`?

**Comment by KristofferC** [comment]
Hmm, first I don't think `vec(A)` should be a `Vec`. Or maybe it should...

If you want a standard array, I think the way should be to just use `Array(A)`. Erroring for `getindex(Tensor, ::Colon)` seems ok.

File: docs/src/man/indexing.md (hunk 1 above)
Comment (KristofferC) [reply docs/src/man/indexing.md:44]: I mean, it is a bit weird already if you enter a non symmetric function to create a symmetric tensor...

**Comment by fredrikekre** [comment]
> Hmm, first I don't think vec(A) should be a Vec. Or maybe it should...

Currently `vec(::Vec)` and `vec(::Tensor)` goes to this in `Base`:

``` jl
vec(a::AbstractArray) = reshape(a,_length(a))
vec(a::AbstractVector) = a
```

We could define our own methods for that perhaps

I also think `Array(A)` is nicer.

File: docs/src/man/indexing.md (hunk 1 above)
Comment (fredrikekre) [reply docs/src/man/indexing.md:44]: fair enough

**Comment by fredrikekre** [comment]
``` jl
julia> A = rand(Tensor{2,3});

julia> B = vec(A)
9-element Base.ReshapedArray{Float64,1,ContMechTensors.Tensor{2,3,Float64,9},Tuple{}}:
 0.299378
 0.265682
 0.452518
 0.737643
 0.627205
 0.316524
 0.203863
 0.221326
 0.961956
```

**Review by fredrikekre — DISMISSED** [review]

File: src/symmetric_ops.jl [hunk 4]
```diff
@@ -215,13 +215,23 @@ julia> A = rand(SymmetricTensor{2,3})
  0.766797  0.460085  0.794026
  0.566237  0.794026  0.854147
 
-julia> eig(A)
-([-0.312033,0.15636,2.06075],
-[0.492843 -0.684993 -0.536554; -0.811724 -0.139855 -0.567049; 0.313385 0.715 -0.624952])
+julia> Λ, Φ = eig(A);
```
Comment (fredrikekre) [inline src/symmetric_ops.jl:218]: this failed, run srand(1234) before making the example

Comment (KristofferC) [reply src/symmetric_ops.jl:218]: oops

File: test/runtests.jl [hunk 5]
```diff
@@ -122,6 +126,8 @@ for T in (Float32, Float64)
                 S = Tensor{order,dim, T}(data)
                 Ssym = symmetric(S)
                 symdata = reshape(Ssym[:],(dim, dim , dim , dim))
```
Comment (fredrikekre) [inline test/runtests.jl]: LUL, first we test if S[:] works and the next line that it throws an error :p

Comment (fredrikekre) [reply test/runtests.jl]: `symdata = Array(Ssym)` works

File: test/runtests.jl [hunk 6]
```diff
@@ -125,7 +125,7 @@ for T in (Float32, Float64)
                 data = rand(T,dim,dim,dim,dim)
                 S = Tensor{order,dim, T}(data)
                 Ssym = symmetric(S)
-                symdata = reshape(Ssym[:],(dim, dim , dim , dim))
```
Comment (KristofferC) [inline test/runtests.jl:124]: thx

**Comment by fredrikekre** [comment]
Merge?

**Pull request merged**